feat(yank): copy formatted query to clipboard on Wayland (wl-copy)

Loures1 opened the pull request:
# Yank Query to Clipboard (Wayland & X11)

This PR adds clipboard support for copying formatted SQL queries.
It automatically uses `wl-copy` on Wayland and `xclip` on X11.

## Changes
- Detects the display server via the environment variable `XDG_SESSION_TYPE`.
- Copies the formatted query (and result) to the clipboard using the appropriate tool.

## How to Use
Select the data menu (**Alt+3**) and press **Y** to copy the query and result to the clipboard.

## Example Output:
<img width="733" height="382" alt="screenshot-2025-10-28_23-34-08" src="https://github.com/user-attachments/assets/83482ddf-caee-4ae8-aa62-e1dc2fc5d4fd" />

<!-- ELLIPSIS_HIDDEN -->

----

> [!IMPORTANT]
> Adds clipboard support for copying formatted SQL queries on Wayland and X11, with new yank actions and popups for confirmation.
> 
>   - **Behavior**:
>     - Adds clipboard support for copying formatted SQL queries using `wl-copy` on Wayland and `xclip` on X11.
>     - Detects display server via `XDG_SESSION_TYPE`.
>     - New keybinding `Y` in `Data` component to trigger yank action.
>   - **Actions**:
>     - Adds `RequestYankData`, `YankData`, and `YankDataFinished` to `Action` enum in `action.rs`.
>   - **Popups**:
>     - Adds `ConfirmYank` popup in `confirm_yank_data.rs` to confirm yank action.
>     - Integrates `ConfirmYank` into `PopUpPayload` in `mod.rs`.
>   - **Components**:
>     - Implements `yank_to_clipboard()` in `data.rs` to handle clipboard operations.
>     - Modifies `handle_key_events()` in `data.rs` to handle `Y` key for yank action.
>   - **Misc**:
>     - Updates help text in `app.rs` to include yank functionality.
> 
> <sup>This description was created by </sup>[<img alt="Ellipsis" src="https://img.shields.io/badge/Ellipsis-blue?color=175173">](https://www.ellipsis.dev?ref=achristmascarl%2Frainfrog&utm_source=github&utm_medium=referral)<sup> for 23371247bb99240d8c8580119e56e851494e83a4. You can [customize](https://app.ellipsis.dev/achristmascarl/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

<!-- ELLIPSIS_HIDDEN -->

achristmascarl commented:
@Loures1 thanks for the PR, and I like the idea of copying the results and the query. Question, does the current clipboard implementation that uses `arboard` work for you on Wayland and X11? i.e., when you go to the history tab, are you able to copy queries, and when you go to the data tab and enter visual mode with `v`, are you able to copy the cells of data?

I ask because the `arboard` implementation currently used should support those platforms, while also being cross platform, so I would prefer to refactor that and use it for this feature as well as opposed to add in new copying mechanisms.

Loures1 commented:
@achristmascarl Ready, I changed it to use arboard and removed the confirm yank popup. The 'Copy All' feature now works like 'Copy Cell' or 'Copy Row'.

Loures1 commented:
@achristmascarl All checks are green. Let me know if any further changes are needed.

achristmascarl commented:
@Loures1, great work! Last piece of feedback is I think we should add the confirmation modal with the number of rows being copied when yanking all back in, to warn users in case they're about to accidentally copy more rows than they intended.

Loures1 commented:
@achristmascarl Done, Now display the confirmation modal with number of rows.
